rplugin/python3/deoplete/sources/flow.py

```diff
@@ -41,20 +42,19 @@ def get_complete_position(self, context):
         return self.__completer.determineCompletionPosition(context)
 
     def gather_candidates(self, context):
-        return self.__completer.find_candidates(
-            context,
-            self.relative()
-        )
+        rel, cfg_dir = self.relative()
+        return self.__completer.find_candidates(context, rel, cfg_dir)
 
     def relative(self):
```

Comment:
I changed `self.relative()` so that it returns a pair of the relative path to the file to be checked, and the path of the Flow config file. (Previously it returned the relative path to the file as a single value).

Put together these changes cache paths for the relevant Flow config for each file that is checked, and pass the config path through to `find_candidates`. I have checked that `self.relative()` returns a pair in all cases. In cases where the plugin cannot locate a config file then a value `None` is given to `Popen` for the `cwd` parameter, which has the same effect as providing no value for that parameter.

Reply:
I see, thank you. I'm not convinced by this function's name but I can't think of anything better.